draft-packages/menu/KaizenDraft/Menu/components/MenuList.spec.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react"
+import { render, screen } from "@testing-library/react"
+import { MenuList, MenuItem } from "../../../"
+
+describe("MenuList", () => {
+  describe("With a heading", () => {
+    it("gives the list an accessible name of the section heading text", () => {
+      render(
+        <MenuList heading={{ children: "A menu section" }}>
+          <MenuItem label="Item one" />
+        </MenuList>
+      )
+      expect(
+        screen.getByRole("list", { name: "A menu section" })
+      ).toBeInTheDocument()
+    })
+
+    it("renders a different element when the tag prop is provided", () => {
+      render(
+        <MenuList heading={{ children: "A menu section", tag: "h2" }}>
+          <MenuItem label="Item one" />
+        </MenuList>
```

Comment:
Have you considered exposing the heading component and have consumers compose it instead?

```
<MenuList>
  <MenuListHeading id="list-heading-id" tag="h2">A menu section</MenuListHeading>
  <MenuItem label="Item one" />
</MenuList>
```

Would also simplify your `listHeadingId` implementation.

Reply:
Ok just explored that option here: https://github.com/cultureamp/kaizen-design-system/pull/3604

It's a little different from your code sample, the menu needs to sit outside of the list because we can't have it sitting inside the `<ul>`. The MenuList also needs to link to the heading id with `aria-labelledby`